### DISCUSSION TITLE
cuda steam 功能 使我原来20fps 跃升到了 最高 40fps   我是4090 转1080分辨率的视频

### DISCUSSION THREAD

**Discussion by hanweinjq** [nagadomi/nunif#217] [General]
cuda steam 功能 使我原来20fps 跃升到了 最高 40fps   我是4090 转1080分辨率的视频
![image](https://github.com/user-attachments/assets/78646866-35a6-4d37-83a7-66e22a047c0f)
提一点改进意见 就是cuda steam显存销毁的机制还需要优化下，我发现会不断积累显存到满 然后才销毁 希望能限定一个值比如 22G，到了就销毁。感谢作者的贡献！

**Comment by nagadomi** [comment]
That's good to know.
If the output video sometimes results in random noise, there may be a bug in CUDA stream related code.

**Reply by hanweinjq** [reply]
我目前没有遇到这种情况，fps 从不勾选的 24fps 提升到了 40fps左右，这是很大的改进！感谢作者！ 目前唯一遇到的问题就是显存会不断的增长到满，满的时候 fps会降低一段时间 然后显存回收之后 fps会回来。

另外我喜欢批量处理短视频，多个文件夹的视频 如果能显示总体的处理的进度 将会方便很多！

**Reply by nagadomi** [reply]
It is normal behavior of Shared GPU Memory on Windows.
good explanation: https://www.reddit.com/r/StableDiffusion/comments/x308i4/comment/immq838/
